multicluster/apis/multicluster/v1alpha1/multiclusterconfig_types.go

```diff
@@ -21,13 +21,26 @@ import (
 	config "sigs.k8s.io/controller-runtime/pkg/config/v1alpha1"
 )
 
+// Precedence defines the precedence of Node IP type.
+type Precedence string
+
+const (
+	PrecedencePrivate = "private"
+	PrecedencePublic  = "public"
+)
+
 //+kubebuilder:object:root=true
 
 // MultiClusterConfig is the Schema for the multiclusterconfigs API
 type MultiClusterConfig struct {
 	metav1.TypeMeta `json:",inline"`
 	// ControllerManagerConfigurationSpec returns the contfigurations for controllers
 	config.ControllerManagerConfigurationSpec `json:",inline"`
+	// ServiceCIDR allows user to set the Cluster IP range of the cluster manually.
+	ServiceCIDR string `json:"serviceCIDR,omitempty"`
+	// The precedence about which IP (private or public one) of Node is preferred to
+	// be used as tunnel endpoint. If it's empty, private IP will be chose.
+	Precedence Precedence `json:"precedence,omitempty"`
```

Comment:
The gateway IPs are provided by user or auto-discovered from Node? Do we support user configured IP?

Reply:
It's auto-discovered by checking Node's status which has ExternalIP or InternalIP or both.
do we need to support configured IP?

Reply:
How TunnelEndpoint is created? It is by user and then MC controller adds Node IPs to it? In that case, should MC controller update Spec directly?

BTW, I feel we should just name the CRD GatewayNode, or even MultiClusterGateway to be cleaer. TunnelEndpoint is too generic.

Reply:
TunnelEndpoint is created by MC controller automatically when user annotate a Node with `multicluster.antrea.io/gateway:true`, I will rename it.

Reply:
Do you think we can have users create the TunnelEndpoint CR, esp. if we want to enable more configuration later (e.g. load sharing, separate interface/IP)? That is one reason I hope to understand how to do load balancing or select active instance with multiple gateways.

Reply:
yes, user can definitely create the TunnelEndpoint CR, I can remove current TunnelEndpoint webhook, then user can create it manually.

Reply:
We should name it "GatewayIPPrecedence" to be clear.